### ISSUE TITLE
Add .editorconfig to Serverless Project

### ISSUE THREAD

**Issue by erikerikson** [serverless/serverless#2180]
# This is a Feature (rly: Convenience) Proposal

## Description

Writing code for Serverless if your default IDE settings do not match those of the Serverless project and/or lint causes one to delete extra spaces, accidentally check in white space changes and other silly and undesirable things.

I propose using an .editorconfig file makes these sorts of things go away.  See http://editorconfig.org/ for details.  An example .editorconfig file might look like (not sure I've quite included all details from the lint):

```
root = true  ; top-most EditorConfig file

; Unix-style newlines with a newline ending every file
[*]
charset = utf-8
end_of_line = lf
indent_size = 2
indent_style = space
insert_final_newline = true
trim_trailing_whitespace = true
```

**Comment by flomotlik** [comment]
Sounds good to me, can you open a PR with the first implementation there and add a note to the Contribution Guide to the Code Style section about it?

**Comment by flomotlik** [comment]
implemented with #2192
